### PR TITLE
slimes blood colour is their body colour

### PR DESCRIPTION
**Pull request by timothyteakettle**
## About The Pull Request
this adds in support for custom blood colours by having your blood colour be defined by a var in your species datum
it also makes it so when updating a character's body, if the character is a slime, their blood colour is updated to their body colour

## Why It's Good For The Game
better character customization is good

## Changelog
:cl:
add: support for custom blood colours implemented, slimes blood colour now equivalent to their body colour
/:cl:

### CODE REVIEW

**Comment by Ragolution** [comment]
Does this effect the coloration of blood bags at all?

File: code/modules/mob/living/carbon/update_icons.dm [hunk 1]
```diff
@@ -193,6 +193,9 @@
 
 /mob/living/carbon/update_body()
 	update_body_parts()
+	// snowflakey thing: update blood colour if a slimeperson
+	if(isjellyperson(src))
+		dna.species.exotic_blood_color = "#" + dna.features["mcolor"]
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/update_icons.dm]: Don't you have anywhere else (like species/proc/handle_mutant_bodyparts) to move this string of code to?

**Review by Ghommie — DISMISSED** [review]

File: code/modules/mob/living/carbon/update_icons.dm (hunk 1 above)
Comment (timothyteakettle) [reply code/modules/mob/living/carbon/update_icons.dm]: ill try moving it into there, didn't think about that

**Comment by timothyteakettle** [comment]
also before this is merged I want to attempt to fix a blood-related issue with clothing as it probably fits within the scope of all the other blood-related changes i'm making here

**Comment by timothyteakettle** [comment]
also in reply about blood bags, no, this does not affect them currently, they'll fill up using the default blood colours
I could probably make it so it updates the colours to whatever the most recent blood added is though, or attempt to make it blend the existing colours

edit: I stand corrected it does change the colour of blood bags!

File: code/modules/mob/living/carbon/update_icons.dm (hunk 1 above)
Comment (timothyteakettle) [reply code/modules/mob/living/carbon/update_icons.dm]: placed it in the handle_mutant_bodyparts method for jellypeople

**Comment by timothyteakettle** [comment]
it now applies the correct colour when splashing blood from beakers onto turfs, and splashing blood onto clothing

**Review by Ghommie — APPROVED** [review]

**Pull request merged**